### PR TITLE
Cypress e2e + Github Actions + Hardhat

### PR DESCRIPTION
**Pull request by rafinskipg**
## What does this PR do?

- Cypress allows us to write e2e tests and see the screen / steps . To easily visualize what is happening

- Hardhat allows us to launch a goerli fork at a certain block 

### Running the network in hardhat mode
```
yarn run hardhat
yarn run dev   -> Access to http://localhost:3000/polling?network=goerlifork
```

## Tasks
- Set up secret enviroment variables for github actions
- Create github actions 
  - Lint 
  - Build
  - Test Unit
  - Test cypress 
  - Run codecoverage
 - Record Tests on https://dashboard.cypress.io/projects/uckcr1/runs 
 - Connect account with from the e2e tests
 - Delete previous e2e tests
 - Use hardhat to fork goerli and deprecate testchain
 
 
 ## Future improvements
 - Remove the testchain dependency
 - Refactor the unit tests to not use testchain

### CODE REVIEW

File: hardhat.config.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  networks: {
+    localhost: {
+      url: 'http://127.0.0.1:8545',
+      timeout: 2000000,
+      chainId: 31337
+    },
+
+    hardhat: {
+      forking: {
+        url: 'https://eth-goerli.alchemyapi.io/v2/p7bY4ggxW60weHKPiAP7HXN2RNAAQZ8E',
```
Comment (adamgoth) [inline hardhat.config.js:21]: Should this key be an `env` var?

File: modules/contracts/constract.constants.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+export const GOERLI_ADDRESSES = {
```
Comment (adamgoth) [inline modules/contracts/constract.constants.ts:1]: There's a typo in `contract` in this file name

File: modules/web3/hooks/useWindowbindings.tsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
+import getMaker from 'lib/maker';
+import { useEffect, useState } from 'react';
+
+type PrivateKeyAccount = {
+  address: string;
+  key: string;
+};
+
+export function useWindowBindings(): void {
```
Comment (adamgoth) [inline modules/web3/hooks/useWindowbindings.tsx:9]: Small nit pick but casing in file name should match function name?

File: pages/polling/review.tsx [hunk 4]
```diff
@@ -85,26 +85,45 @@ const PollingReview = ({ polls }: { polls: Poll[] }) => {
...
+                        </PollOverviewCard>
+                      );
+                    })}
+                  </Stack>
+                )}
+                {votedPolls.length === 0 && (
+                  <Box>
+                    <Text>There are no polls added to your ballot.</Text>
+                  </Box>
+                )}
```
Comment (adamgoth) [inline pages/polling/review.tsx:115]: Hm, I thought I already added this to develop, not sure why it's showing as a change in your PR 🤔

File: hardhat.config.js (hunk 1 above)
Comment (rafinskipg) [reply hardhat.config.js:21]: yeah probably

File: pages/polling/review.tsx (hunk 4 above)
Comment (rafinskipg) [reply pages/polling/review.tsx:115]: idk 🤷

**Review by b-pmcg — APPROVED** [review]
Nice, this is amazing. A couple tests failed for me due to timeout and my computer fans sound like it's about to go orbital, but LGTM

**Review by adamgoth — APPROVED** [review]

**Pull request merged**